client/app/lib/util/routeHandler.coffee

```diff
@@ -50,10 +50,51 @@ module.exports = (options) ->
 
 
   lazyrouter.bind name, (type, info, state, path, ctx) ->
+
+    modalContainer = new ModalContainer
+
     switch type
+
+      when 'group-disabled'
+
+        { role, status } = info.params
+
+        { appManager } = kd.singletons
+
+        return handleDisabled role, status  if frontApp = appManager.getFrontApp()
+
+        appManager.once 'AppIsBeingShown', -> handleDisabled role, status
+
+        appManager.open 'IDE', { forceNew: yes }, (app) ->
+          app.amIHost = yes
+          appManager.tell 'IDE', 'showNoMachineState'
```

Comment:
it might not enough to prevent IDE access, we may need to put this check on kloud and klient kite implementations for front-end.